### PR TITLE
[SKOOP-111] Displaying information for a 'deletion of a community' notification

### PR DESCRIPTION
**Pull request by svetivanova**
displaying information for a 'deletion of a community' notification
add more notification types
tests
@LebedkoDmitry could you please take a look?

### CODE REVIEW

File: src/app/my-messages/member-left-community-notification.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { AbstractNotification } from './abstract-notification';
 import { Community } from '../communities/community';
+import { User } from '../users/user';
 
 export class MemberLeftCommunityNotification extends AbstractNotification {
   community: Community;
+  user: User;
```
Comment (LebedkoDmitry) [inline src/app/my-messages/member-left-community-notification.ts]: Is it necessary to add this field? Are we going to remove this field in future as it does not make sense for the authenticated user.

File: src/app/my-messages/my-messages.component.spec.ts [hunk 2]
```diff
@@ -234,4 +277,56 @@ describe('MyMessagesComponent', () => {
...
+
+    // MEMBER_KICKED_OUT_OF_COMMUNITY -> link to the community from a notification object
+    communityRow = getCommunityInformation(notificationCards[5]);
+    actualData = communityRow.nativeElement.querySelector('a');
+    expect(actualData).toBeDefined();
+    expect(actualData.href).toContain(`/communities/${expectedNotifications[5].community.id}`);
+    expect(actualData.text).toBe(expectedNotifications[5].community.title);
+  }));
+
+  function getCommunityInformation(notificationCard: DebugElement): DebugElement {
```
Comment (LebedkoDmitry) [inline src/app/my-messages/my-messages.component.spec.ts]: I suggest to define this function in the scope of the test `should display community information regarding to the notification type` to emphasize it is used only there. What do you think?

File: src/app/my-messages/my-messages.component.spec.ts [hunk 3]
```diff
@@ -234,4 +277,56 @@ describe('MyMessagesComponent', () => {
...
   }));
+
+  it('should display community information regarding to the notification type', fakeAsync(() => {
+    fixture.detectChanges();
+
+    const notificationCards = fixture.debugElement.queryAll(By.css(('.messages-card')));
+    let communityRow = getCommunityInformation(notificationCards[0]);
+
+    // INVITATION_TO_JOIN_COMMUNITY -> link to the community from a registration object
+    let actualData = communityRow.nativeElement.querySelector('a');
```
Comment (LebedkoDmitry) [inline src/app/my-messages/my-messages.component.spec.ts]: I suggest to rename `actualData` to `linkToCommunity` to show what object we really work with.

File: src/app/my-messages/my-messages.component.spec.ts [hunk 4]
```diff
@@ -234,4 +277,56 @@ describe('MyMessagesComponent', () => {
...
+    expect(actualData).toBeDefined();
+    expect(actualData.href).toContain(`/communities/${expectedNotifications[0].registration.community.id}`);
+    expect(actualData.text).toBe(expectedNotifications[0].registration.community.title);
+
+    // REQUEST_TO_JOIN_COMMUNITY -> link to the community from a registration object
+    communityRow = getCommunityInformation(notificationCards[1]);
+    actualData = communityRow.nativeElement.querySelector('a');
```
Comment (LebedkoDmitry) [inline src/app/my-messages/my-messages.component.spec.ts]: I suggest to rename `actualData` to `linkToCommunity` to show what object we really work with.

File: src/app/my-messages/my-messages.component.spec.ts [hunk 5]
```diff
@@ -234,4 +277,56 @@ describe('MyMessagesComponent', () => {
...
+    expect(actualData).toBeDefined();
+    expect(actualData.href).toContain(`/communities/${expectedNotifications[0].registration.community.id}`);
+    expect(actualData.text).toBe(expectedNotifications[1].registration.community.title);
+
+    // ACCEPTANCE_TO_COMMUNITY -> link to the community from a registration object
+    communityRow = getCommunityInformation(notificationCards[2]);
+    actualData = communityRow.nativeElement.querySelector('a');
```
Comment (LebedkoDmitry) [inline src/app/my-messages/my-messages.component.spec.ts]: I suggest to rename `actualData` to `linkToCommunity` to show what object we really work with.

File: src/app/my-messages/my-messages.component.spec.ts [hunk 6]
```diff
@@ -234,4 +277,56 @@ describe('MyMessagesComponent', () => {
...
+    expect(actualData).toBeDefined();
+    expect(actualData.href).toContain(`/communities/${expectedNotifications[1].registration.community.id}`);
+    expect(actualData.text).toBe(expectedNotifications[2].registration.community.title);
+
+    // COMMUNITY_DELETED -> community name
+    communityRow = getCommunityInformation(notificationCards[3]);
+    actualData = communityRow.nativeElement.innerText;
```
Comment (LebedkoDmitry) [inline src/app/my-messages/my-messages.component.spec.ts]: I suggest to rename `actualData` to `communityName` to show what object we really work with.

File: src/app/my-messages/my-messages.component.spec.ts [hunk 7]
```diff
@@ -234,4 +277,56 @@ describe('MyMessagesComponent', () => {
...
+    // MEMBER_LEFT_COMMUNITY -> link to the community from a notification object
+    communityRow = getCommunityInformation(notificationCards[4]);
+    actualData = communityRow.nativeElement.querySelector('a');
+    expect(actualData).toBeDefined();
+    expect(actualData.href).toContain(`/communities/${expectedNotifications[4].community.id}`);
+    expect(actualData.text).toBe(expectedNotifications[4].community.title);
+
+    // MEMBER_KICKED_OUT_OF_COMMUNITY -> link to the community from a notification object
+    communityRow = getCommunityInformation(notificationCards[5]);
+    actualData = communityRow.nativeElement.querySelector('a');
```
Comment (LebedkoDmitry) [inline src/app/my-messages/my-messages.component.spec.ts]: I suggest to rename `actualData` to `linkToCommunity` to show what object we really work with.

File: src/app/my-messages/member-left-community-notification.ts (hunk 1 above)
Comment (svetivanova) [reply src/app/my-messages/member-left-community-notification.ts]: We talked about MemberKickedOutNotification 😉 
But MemberLeftCommunityNotification must contain a user to show manager which user has left a community.

File: src/app/my-messages/my-messages.component.spec.ts (hunk 2 above)
Comment (svetivanova) [reply src/app/my-messages/my-messages.component.spec.ts]: Thank you. Done.

File: src/app/my-messages/my-messages.component.spec.ts (hunk 3 above)
Comment (svetivanova) [reply src/app/my-messages/my-messages.component.spec.ts]: Thank you. Done.

File: src/app/my-messages/my-messages.component.spec.ts (hunk 4 above)
Comment (svetivanova) [reply src/app/my-messages/my-messages.component.spec.ts]: Thank you. Done.

File: src/app/my-messages/my-messages.component.spec.ts (hunk 5 above)
Comment (svetivanova) [reply src/app/my-messages/my-messages.component.spec.ts]: Thank you. Done.

File: src/app/my-messages/my-messages.component.spec.ts (hunk 6 above)
Comment (svetivanova) [reply src/app/my-messages/my-messages.component.spec.ts]: Thank you. Done.

File: src/app/my-messages/my-messages.component.spec.ts [hunk 8]
```diff
@@ -234,4 +277,56 @@ describe('MyMessagesComponent', () => {
...
+
+    // COMMUNITY_DELETED -> community name
+    communityRow = getCommunityInformation(notificationCards[3]);
+    actualData = communityRow.nativeElement.innerText;
+    expect(actualData).toBeDefined();
+    expect(actualData).toBe(expectedNotifications[3].communityName);
+
+    // MEMBER_LEFT_COMMUNITY -> link to the community from a notification object
+    communityRow = getCommunityInformation(notificationCards[4]);
+    actualData = communityRow.nativeElement.querySelector('a');
```
[the remark this answers is not in the document]
Comment (svetivanova) [reply src/app/my-messages/my-messages.component.spec.ts]: Thank you. Done.

File: src/app/my-messages/my-messages.component.spec.ts (hunk 7 above)
Comment (svetivanova) [reply src/app/my-messages/my-messages.component.spec.ts]: Thank you. Done.

File: src/app/my-messages/member-left-community-notification.ts (hunk 1 above)
Comment (LebedkoDmitry) [reply src/app/my-messages/member-left-community-notification.ts]: Yes, you are right.

**Comment by LebedkoDmitry** [comment]
Approved.

**Pull request merged**